File: #1155.number-of-dice-rolls-with-target-sum.cpp
class Solution {
public:
    int numRollsToTarget(int n, int k, int target) {
    
        vector<int> v(target + 1);
        v[0] = 1;
        for (int i = 0; i < n; i++) {
            vector<int> cur(target + 1);
            for (int j = 1; j <= target; j++) { 
                for(int f = 1; f <= std::min(k, j); f++) { 
                    cur[j] = (cur[j] + v[j - f]) % (int)(1e9 + 7);
                }
            }
            v = cur;
        }
        return v.back();
    }
};

File: #1155.number-of-dice-rolls-with-target-sum.js
/**
 * @param {number} n
 * @param {number} k
 * @param {number} target
 * @return {number}
 */
var numRollsToTarget = function(n, k, target) {
    const mod = 10**9 + 7;
    const dp = [...Array(target+1)].map(
      () => Array(n + 1).fill(0)
    );
    dp[0][0] = 1;
    
    for (let i = 1; i <= target; i++) {
        for (let j = 1; j <= n; j++) {
            for (let l = 1; l <= k; l++) {
                if (l > i) {
                    break;
                }
                dp[i][j] += dp[i - l][j - 1] % mod;                
            }
        }
    }
    return dp[target][n] % mod;
};
